### PR TITLE
Binary tree level order traversal

### PR DESCRIPTION
**Pull request by gaizzzzi**

### CODE REVIEW

File: main/level_order_traversal.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class Tree:
+    def build_tree(self, l) -> TreeNode:
```
Comment (sheep-sheep) [inline main/level_order_traversal.py:10]: `l` -> `input_list` can improve the readability.

File: main/level_order_traversal.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        if not list:
```
Comment (sheep-sheep) [inline main/level_order_traversal.py:11]:
```
if not input_list:
```

File: main/level_order_traversal.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            return None
+
+        hashmap = {}
```
Comment (sheep-sheep) [inline main/level_order_traversal.py:14]: Python doesn't use "Hashmap", it's a Java concept, this is a `dict`

Comment (gaizzzzi) [reply main/level_order_traversal.py:14]: how should I name it to increase the readability?

File: main/level_order_traversal.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            i += 1
+        return hashmap[0]
+            
+            
+    def level_order(self, root: TreeNode) -> [[]]:
+        if not root:
+            return []
+        queue = [root]
+        level_order_vals = []
+        while queue:
```
Comment (sheep-sheep) [inline main/level_order_traversal.py:42]: queue keeps changing during the while loop, I thought the compiler should complain about it.

File: main/level_order_traversal.py (hunk 2 above)
Comment (gaizzzzi) [reply main/level_order_traversal.py:11]: got it

File: main/level_order_traversal.py (hunk 1 above)
Comment (gaizzzzi) [reply main/level_order_traversal.py:10]: good catch thx ^ ^

File: main/level_order_traversal.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            temp = []
+            level = []
+            for q in queue:
+                level.append(q.val)
+                if q.left:
+                    temp.append(q.left)
+                if q.right:
+                    temp.append(q.right)
+            queue = temp
+            level_order_vals.append(level)
```
Comment (sheep-sheep) [inline main/level_order_traversal.py:52]:
```
n = len(queue)
while n:
    level.append(queue.pop(0).val)
    if q.left:
        queue.append(q.left)
    ...
```

**Pull request closed without merging**